lms/static/scripts/frontend_apps/components/dashboard/StudentStatusBadge.tsx

```diff
@@ -0,0 +1,27 @@
+import classnames from 'classnames';
+
+export type StudentStatusType = 'new' | 'error' | 'syncing' | 'drop';
+
+export default function StudentStatusBadge({
+  type,
+}: {
+  type: StudentStatusType;
+}) {
+  return (
+    <div
+      className={classnames(
+        'px-1 py-0.5 rounded cursor-auto font-bold uppercase text-[0.65rem]',
+        {
+          'bg-grey-7 text-white': type === 'new' || type === 'drop',
+          'bg-grade-error-light text-grade-error': type === 'error',
+          'bg-grey-2 text-grey-7': type === 'syncing',
+        },
+      )}
+    >
+      {type === 'new' && 'New'}
+      {type === 'error' && 'Error'}
+      {type === 'syncing' && 'Syncing'}
+      {type === 'drop' && 'Drop'}
+    </div>
```

Comment:
Can this badge live update while the page is loaded? If so we may need to consider accessibility affordances for this. Leave that out of scope for this PR though.

Reply:
The ones related with grades sync can update at runtime.

When clicking sync grades, all badges showing "new" or "error" will transition into syncing, and once finished, they will either disappear or transition back to error.

I'll create a separate issue to wrap them into an `aria-live` container.